ui/app/models/task.js

```diff
@@ -50,4 +50,27 @@ export default class Task extends Fragment {
   @attr('number') reservedEphemeralDisk;
 
   @fragmentArray('volume-mount', { defaultValue: () => [] }) volumeMounts;
+
+  async _fetchParentJob() {
+    let job = await this.store.findRecord('job', this.taskGroup.job.id, {
+      reload: true,
+    });
+    this._job = job;
+  }
+
+  get pathLinkedVariable() {
+    if (!this._job) {
+      this._fetchParentJob();
+      return null;
+    } else {
+      let jobID = this._job.plainId;
+      if (this._job.parent.get('plainId')) {
+        jobID = this._job.parent.get('plainId');
+      }
+      return this._job.variables?.findBy(
+        'path',
+        `jobs/${jobID}/${this.taskGroup.name}/${this.name}`
+      );
+    }
+  }
```

Comment:
`question`:  How will we handle this behavior in our Mirage factory?

Reply:
Mirage factory is pretty simple right now: it just returns an unfiltered list of our vars at /vars. We don't attempt to mimic the filter behaviour because of the potential for complexity here, and because the `pathLinkedVariable` equality check works with a filtered or unfiltered variables list.